990 introduce support for the tdls library (#992)
thelfer/tfel: docs/mfront/LinearSystemSolver.md

Reviewer: oot_threshold and singular_eps are not listed here even though theyre important

Author: Ok, I just though they could be left as internal paramters. I will add them

> The `@LinearSystemSolver` keyword is used to select the linear sytem

Reviewer: ```suggestion
The `@LinearSystemSolver` keyword is used to select the linear system
```

> solver. It is followed by the name of solver. Currently, two solvers

Reviewer: ```suggestion
solver. It is followed by the name of the solver. Currently, two solvers
```

are supported:

- `Default`. This uses the solvers provided natively by the `TFEL/Math`
  library. As implied by its name, this solver is used by default.
- `TDLS` (or equivalently `TinyDeviceCallableLinearSolver`). See
  <https://trsxvz.github.io/TDLS/getting_started.html> for details. If

>   `TFEL` is not compiled with support of the `TDLS` library selecting

Reviewer: ```suggestion
  `TFEL` is not compiled with support for the `TDLS` library, selecting
```

>   this solver falls back the the `Default` Solver.

Reviewer: ```suggestion
  this solver falls back to the `Default` solver.
```

> The name of the solver can optionnaly be followed by a dictionary

Reviewer: ```suggestion
The name of the solver can optionally be followed by a dictionary
```

allowing to parametrize the solver.

## Supported Parameters

### Parameters of the `Default` Linear Solver

The `Default` Linear Solver does not accept any parameter.

### Parameters of the `TDLS` Linear Solver

The following options are supported:

- `tile_size`: the matrix is processed as a grid of `tile_size x
  tile_size` register tiles. This is the main performance axis of the
  solvers - tune it per system dimension. The `tile_size` can be a

>   strictly positive integer or a string. In `tile_size` is a string, it

Reviewer: ```suggestion
  strictly positive integer or a string. If `tile_size` is a string, it
```

>   is interpreted as a integer formula whose variable can be any integer

Reviewer: ```suggestion
  is interpreted as an integer formula whose variables can be any integer
```

  constant (see the `@IntegerConstant` keyword), `N`, `TVectorSize`,
  `StensorSize` or `TensorSize`. This formula must evaluate to a
  strictly positive integer for all hypotheses supported by the
  behaviour.
- `schedule`: expected values are either `left_looking` (or equivalently

>   `left`, `left-looking` or `LeftLooking`) and `right_looking` (or

Reviewer: ```suggestion
  `left`, `left-looking` or `LeftLooking`) or `right_looking` (or
```

  equivalently `right`, `right-looking` or `RightLooking`). This
  parameter defines the elimination schedule of the tiled factorization.
- `out_of_tile_search_strategy`: expected values are either
  `first_acceptable` (or equivalently `first-acceptable` or
  `FirstAcceptable`) or `full_scan` (or equivalently `full-scan` or
  `FullScan`).

> - unroll_inner_loop: expected values are either `true` or `false`. If

Reviewer: ```suggestion
- `unroll_inner_loop`: expected values are either `true` or `false`. If
```
